app/src/main/res/layout/activity_edit_profile.xml

```diff
@@ -14,77 +14,82 @@
+                android:contentDescription="@null"
+                app:layout_constraintTop_toTopOf="parent" />
 
+            <ImageButton
+                android:id="@+id/headerButton"
+                android:layout_width="match_parent"
+                android:layout_height="200dp"
+                android:background="#66000000"
+                android:contentDescription="@string/label_header"
+                app:layout_constraintTop_toTopOf="parent"
+                app:srcCompat="@drawable/ic_add_a_photo_32dp" />
+
+            <ProgressBar
+                android:id="@+id/headerProgressBar"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:indeterminate="true"
+                android:visibility="gone"
+                app:layout_constraintBottom_toBottomOf="@id/headerPreview"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <com.keylesspalace.tusky.view.RoundedImageView
+                android:id="@+id/avatarPreview"
+                android:layout_width="80dp"
+                android:layout_height="80dp"
+                android:layout_marginStart="16dp"
+                android:contentDescription="@null"
+                app:layout_constraintBottom_toBottomOf="@id/headerPreview"
+                app:layout_constraintStart_toStartOf="@id/contentContainer"
+                app:layout_constraintTop_toBottomOf="@id/headerPreview" />
+
+            <ImageButton
+                android:id="@+id/avatarButton"
+                android:layout_width="80dp"
+                android:layout_height="80dp"
+                android:layout_marginStart="16dp"
+                android:background="@drawable/round_button"
+                android:contentDescription="@string/label_avatar"
+                android:elevation="4dp"
+                app:layout_constraintBottom_toBottomOf="@id/headerPreview"
+                app:layout_constraintStart_toStartOf="@id/contentContainer"
+                app:layout_constraintTop_toBottomOf="@id/headerPreview"
+                app:srcCompat="@drawable/ic_add_a_photo_32dp" />
+
+            <ProgressBar
+                android:id="@+id/avatarProgressBar"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_centerInParent="true"
+                android:layout_marginStart="16dp"
+                android:indeterminate="true"
+                android:visibility="gone"
+                app:layout_constraintBottom_toBottomOf="@id/headerPreview"
+                app:layout_constraintStart_toStartOf="@id/contentContainer"
+                app:layout_constraintTop_toBottomOf="@id/headerPreview" />
 
             <LinearLayout
+                android:id="@+id/contentContainer"
```

Comment:
Do we still need this one? We could put it into `ConstraintLayout` maybe?

Reply:
yes i could, but I think the code is way easier to read with the additional layout so I'd rather leave it